lib/asciidoctor/reader.rb

```diff
@@ -836,11 +836,6 @@ def preprocess_include_directive raw_target, raw_attributes
       if @include_stack.size >= abs_maxdepth
         warn %(asciidoctor: ERROR: #{line_info}: maximum include depth of #{@maxdepth[:rel]} exceeded)
         return
-      elsif ::RUBY_ENGINE_OPAL && ::JAVASCRIPT_IO_MODULE == 'xmlhttprequest'
-        # NOTE resolves uri relative to currently loaded document
-        # NOTE we defer checking if file exists and catch the 404 error if it does not
-        target_type = :file
-        inc_path = relpath = @include_stack.empty? && ::Dir.pwd == @document.base_dir ? target : %(#{@dir}/#{target})
       elsif (Helpers.uriish? target) || ((::URI === @dir) && (target = %(#{@dir}/#{target})))
```

Comment:
The next step is to use `URI.join` instead of `%(#{@dir}/#{target})` so the following will also work:

```ruby
base_dir = 'http://localhost:1234'
target = './name/asciidoctor'
resolved_target = 'http://localhost:1234/name/asciidoctor'
```

```ruby
base_dir = 'http://localhost:1234/name'
target = '../name/asciidoctor'
resolved_target = 'http://localhost:1234/name/asciidoctor'
```

But I don't want to introduce two changes at once 😉

Reply:
To make it work on Asciidoctor.js we will also need to implement `URI.join` in Opal.